homeassistant/components/nina/config_flow.py

```diff
@@ -0,0 +1,163 @@
+"""Config flow for Nina integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from pynina import ApiError, Nina
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import _LOGGER, CONF_FILTER_CORONA, CONF_MESSAGE_SLOTS, CONF_REGIONS, DOMAIN
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for NINA."""
+
+    VERSION: int = 1
+
+    def __init__(self) -> None:
+        """Initialize."""
+        super().__init__()
+        self._all_region_codes_sorted: dict[str, str] = {}
+        self.regions_a: dict[str, Any] = {}
+        self.regions_b: dict[str, Any] = {}
+        self.regions_c: dict[str, Any] = {}
+        self.regions_d: dict[str, Any] = {}
+        self.regions_e: dict[str, Any] = {}
+        self.regions_f: dict[str, Any] = {}
+
+    async def async_step_user(
+        self: ConfigFlow,
+        user_input: dict[str, Any] | None = None,
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, Any] = {}
+
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        has_error: bool = False
+
+        if len(self._all_region_codes_sorted) == 0:
+            try:
+                nina: Nina = Nina()
+
+                self._all_region_codes_sorted = self.swap_key_value(
+                    await nina.getAllRegionalCodes()
+                )
+
+            except ApiError as err:
+                _LOGGER.warning("NINA setup error: %s", err)
+                errors["base"] = "cannot_connect"
+                has_error = True
+            except Exception as err:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception: %s", err)
+                errors["base"] = "unknown"
+                return self.async_abort(reason="unknown")
+
+        if user_input is not None and not has_error:
+            config: dict[str, Any] = user_input
+
+            config[CONF_REGIONS] = []
+
+            if CONF_REGIONS + "1" in user_input:
+                config[CONF_REGIONS] += user_input[CONF_REGIONS + "1"]
+            if CONF_REGIONS + "2" in user_input:
+                config[CONF_REGIONS] += user_input[CONF_REGIONS + "2"]
+            if CONF_REGIONS + "3" in user_input:
+                config[CONF_REGIONS] += user_input[CONF_REGIONS + "3"]
+            if CONF_REGIONS + "4" in user_input:
+                config[CONF_REGIONS] += user_input[CONF_REGIONS + "4"]
+            if CONF_REGIONS + "5" in user_input:
+                config[CONF_REGIONS] += user_input[CONF_REGIONS + "5"]
+            if CONF_REGIONS + "6" in user_input:
+                config[CONF_REGIONS] += user_input[CONF_REGIONS + "6"]
```

Comment:
There is a lot of repetition in your code, it would nice if we can improve this.

Maybe store all regions in a `dict[str, dict[str, Any]]` and improve the grouping algorithm or just use a loop?

I'd also make the numbering more explicit, the numbering makes it kind of implicit.

In `const.py`:

```python
CONST_GROUP_A_TO_D: Final = "_a_to_d"
CONST_GROUP_E_TO_H: Final = "_e_to_h"
CONST_GROUP_I_TO_L: Final = "_i_to_l"
...

CONST_GROUPS: Final = [CONST_GROUP_A_TO_D, CONST_GROUP_E_TO_H, CONST_GROUP_I_TO_L, ...]

```

Then maybe use something like this:

You might even be able to use a generator for it.

```python
for group in CONST_REGION_GROUPS:
  if group_input := user_input.get(CONST_REGIONS + group):
    config[CONF_REGIONS] += group_input
```

Reply:
Why is this resolved?